test/jdk/java/awt/print/Dialog/PrintModalDialog.java

```
@@ -0,0 +1,132 @@
+        frame.getContentPane().setLayout(null);
+        frame.getContentPane().setBackground(Color.pink);
+        jButton1 = new JButton("PRINT");
+        jButton1.addActionListener(e -> jButton1_actionPerformed(e));
+        jButton1.setBounds(new Rectangle(165, 248, 80, 30));
+        frame.getContentPane().add(jButton1, BorderLayout.SOUTH);
+        return frame;
+    }
+
+    static void jButton1_actionPerformed(ActionEvent e) {
+        PrinterJob printJob = null;
+        PageFormat pageFormat = null;
+        Paper prtPaper = null;
+        boolean bPrintFlg = true;
+
+        try {
+            printJob = PrinterJob.getPrinterJob();
+        }
+        catch (SecurityException se) {
+            bPrintFlg = false;
+        }
+
+        if (bPrintFlg) {
+            pageFormat = printJob.pageDialog(printJob.defaultPage());
+            System.out.println("PrintModalDialog: pageFormat = " +
+                    pageFormat.getWidth() / 72.0 + " x " +
+                    pageFormat.getHeight() / 72.0);
+            if (pageFormat != null) {
+                prtPaper = pageFormat.getPaper();
+                pageFormat.setPaper(prtPaper);
+                printJob.setPrintable((g, pf, page) -> {
+                    System.out.println("Calling print");
+                    if (page == 0) {
+                        Graphics2D g2 = (Graphics2D)g;
+                        g2.translate(pf.getImageableX(), pf.getImageableY());
+                        g2.setColor(Color.black);
+                        g2.drawString("Hello World", 20, 100);
+
+                        return Printable.PAGE_EXISTS;
+                    }
+                    return Printable.NO_SUCH_PAGE;
+                }, pageFormat);
+            }
+
+            if (printJob.printDialog()) {
+                try {
+                    printJob.print();
+                }
+                catch (java.awt.print.PrinterException ex) {
+                    ex.printStackTrace();
```

It is better to use PassFailJFrame.forceFail() when a printer exception occurs else the timer keeps running and the error is swallowed even after a printer exception has occurred.

```suggestion
                catch (java.awt.print.PrinterException ex) {
                    ex.printStackTrace();
                    String msg = "PrinterException: " + ex.getMessage();
                    PassFailJFrame.forceFail(msg);
```

---

Thanks for the suggestion. Added as suggested